prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -218,6 +219,84 @@ def canonicalize_dataset_using_columns(
             )
         return datasets.DatasetDict(dataset_dict)
 
+    def canonicalize_dataset_auto(
```

Comment:
This looks very similar to [`canonicalize_dataset_automatically`](https://github.com/neulab/prompt2model/blob/f2eabc1ad0e23ce774b21f48c590bedc40405ff1/prompt2model/dataset_retriever/description_dataset_retriever.py#L439)

Can we merge these two functions?

Reply:
right. this seems to be from the latest merge into main which I havent had a chance to look at. will fix the merge conflict in a way that only one function remains